rasterio/_warp.pyx

```diff
@@ -706,7 +706,11 @@ def _calculate_default_transform(src_crs, dst_crs, width, height,
         raise CRSError(err.errmsg)
 
     except CPLE_AppDefinedError as err:
-        if "Reprojection failed" in str(err) or "Point outside of projection domain" in str(err):
+        if (
+            "Reprojection failed" in str(err) or
+            "Point outside of projection domain" in str(err) or
+            "tolerance condition error" in str(err)
```

Comment:
the error thrower might depends on GDAL/PROJ version

Reply:
Yes, it's possible. "Reprojection failed" is from PROJ 4.9, if I remember correctly.